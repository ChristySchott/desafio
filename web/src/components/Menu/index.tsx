import React from 'react';

import { useCategory } from 'hooks/category';
import Container from './styles';

interface MenuProps {
  open: boolean;
}

const Menu: React.FC<MenuProps> = ({ open }) => {
  const { categoriesList, setCategory } = useCategory();

  return (
    <Container open={open}>
      <ul>
        <li>
          <a href="/">PÃ¡gina inicial</a>
        </li>
        {categoriesList &&
          categoriesList.map(({ name, path }) => (
            <li key={name}>
              <button type="button" onClick={() => setCategory(path)}>
                {name}
              </button>
            </li>
          ))}
        <li>
          <a href="/cart">Carrinho</a>
        </li>
      </ul>
    </Container>
  );
};
export default Menu;
