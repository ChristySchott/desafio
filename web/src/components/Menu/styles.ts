import styled from 'styled-components';

interface ContainerProps {
  open: boolean;
}

const Container = styled.nav<ContainerProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  background: #c81120;
  height: 100vh;
  text-align: left;
  padding: 2rem;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  transform: ${({ open }) => (open ? 'translateX(0)' : 'translateX(-100%)')};
  transition: transform 0.3s ease-in-out;

  @media (max-width: 320px) {
    width: 100%;
  }

  div {
    text-align: center;
  }

  a {
    display: block;
    font-size: 2rem;
    text-transform: uppercase;
    padding: 2rem 0;
    font-weight: bold;
    letter-spacing: 0.5rem;
    color: #fff;
    text-decoration: none;
    transition: color 0.3s linear;

    @media (max-width: 320px) {
      font-size: 1.5rem;
      text-align: center;
    }

    &:hover {
      color: #ddd;
    }
  }
`;

export default Container;
