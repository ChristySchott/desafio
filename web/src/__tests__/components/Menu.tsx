import React from 'react';
import { render } from '@testing-library/react';

import Menu from '../../components/Menu';

describe('<Menu />', () => {
  it('should render with no error', () => {
    const { container } = render(<Menu open={false} />);

    expect(container).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
