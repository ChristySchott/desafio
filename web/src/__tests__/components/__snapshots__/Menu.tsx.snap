// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render with no error 1`] = `
<nav
  class="sc-bdnylx ebzUpE"
>
  <ul>
    <li>
      <a
        href="/"
      >
        PÃ¡gina inicial
      </a>
    </li>
    <li>
      <a
        href="/"
      >
        Contato
      </a>
    </li>
  </ul>
</nav>
`;
